bundler/helpers/v2/spec/functions_spec.rb

```diff
@@ -21,4 +21,49 @@
       end
     end
   end
+
+  describe "#git_specs" do
+    let(:project_name) { "git_source" }
+    subject(:git_specs) do
+      in_tmp_folder do
+        Functions.git_specs(
+          dir: tmp_path,
+          gemfile_name: "Gemfile",
+          credentials: {}
+        )
+      end
+    end
+
+    def expect_specs(count)
+      expect(git_specs.size).to eq(count)
+      git_specs.each do |gs|
+        auth_uri = gs[:auth_uri]
+        expect { URI.parse(auth_uri) }.not_to raise_error
+        expect(auth_uri).to start_with("https://")
```

Comment:
[For the sake of consistency](https://github.com/dependabot/dependabot-core/pull/3911/commits/6a448360757a74ba0897a71f3cc6f13a9ac7c75e#diff-8fa2e59ed1b104d83c365010bcd2db2ed24eeb58b3136bf9fc5df358bf7a7ad2R178), shouldn't `expect_specs` also be okay with `auth_uri` starting with `http://`?

Reply:
I opted to keep the test simple, and update any `http://` fixtures to prefer https - https://github.com/dependabot/dependabot-core/pull/3911/files#diff-edfa8c988cb28638a8cbce2d2324ac53cc8efefd0e38721156591a13d97ad031R10 .

I'd like to stick there: it's 2021, everything should be `https://`. Since the test is driven by fixtures, it can be updated if support for `http://` becomes an important feature.

Reply:
This was a good shout: the fixture was easy to fix for _this_ test, but it's re-used by others.

Untangling became such a mess that I opted to do `http://` and align this expectation with the implementation.